### PR TITLE
Make Bullet into a GameObject so Timer and Groups don't need to be passed to it.

### PR DESCRIPTION
**Pull request by dvirk**
This deals with issue #72

### CODE REVIEW

File: src/model.py [hunk 1]
```diff
@@ -19,6 +19,13 @@ def __new__(cls) -> _GroupsBase:
         args += [LayeredUpdates()]
         return super(Groups, cls).__new__(cls, *args)  # type: ignore
 
+    def empty(self) -> None:
```
Comment (mick-warehime) [inline src/model.py]: what does empty do? comment?

File: src/test/test_humanoid.py [hunk 2]
```diff
@@ -83,13 +88,7 @@ def _dist(pos_0: Vector2, pos_1: Vector2) -> float:
...
 class ModelTest(unittest.TestCase):
     def tearDown(self) -> None:
-        groups = Connection.groups
-        groups.walls.empty()
-        groups.mobs.empty()
-        groups.bullets.empty()
-        groups.all_sprites.empty()
-        groups.items.empty()
-
+        Connection.groups.empty()
```
Comment (mick-warehime) [inline src/test/test_humanoid.py]: what is a connection object? why not just a class member at setup time?

File: src/model.py (hunk 1 above)
Comment (dvirk) [reply src/model.py]: It calls empty() on all the Group fields. I'll add a docstring.

File: src/test/test_humanoid.py (hunk 2 above)
Comment (dvirk) [reply src/test/test_humanoid.py]: Done

**Pull request merged**